### ISSUE TITLE
SSH pub key auth, agents and agent forwarding, session reuse

### ISSUE THREAD

**Issue by willfurnass** [rcgsheffield/sheffield_hpc#1009]
Need to add info re best practices, esp. re key mgmt/encryption.

**Comment by willfurnass** [comment]
Good tips in this article that featured on Hacker News today -  https://smallstep.com/blog/ssh-agent-explained/:
 - Explains how agent forwarding can by hijacked
 - Recommends selectively enabling agent forwarding so you only have it turned on for specific connections to specific machines you trust,
 - Or one could use `ProxyJump` instead 
---
@ptheywood: Provide examples of how to generate keys with passphrases (or point to an external resource which provides this)?
---
@ptheywood: Recommending limiting the use of a given ssh key-pair - much like having separate passwords on different website you can have separate keys for different remote services (or different local machines, with pub keys)?

**Comment by willfurnass** [comment]
May also want to mention SSH session reuse:
```
$ cat ~/.ssh/config
...
ControlMaster auto
ControlPath ~/.ssh/sockets/%r@%h-%p
ControlPersist 600
...
```

**Comment by willfurnass** [comment]
Closed via #1028
